File: Cpp/1742_Maximum_Number_of_Balls_in_a_Box.cpp
class Solution {
public:
    int countBalls(int lowLimit, int highLimit) {
        int result = 0;
        vector<int> count(46);

        for(int i = lowLimit; i <= highLimit; ++i) {
            int sum = 0;
            int n = i;

            while(n) {
                sum += n % 10;
                n /= 10; 
            }

            result = max(result, ++count[sum]);
        }
        
        return result;
    }
};

File: Kotlin/1742_Maximum_Number_of_Balls_in_a_Box.kt
class Solution {
    fun countBalls(lowLimit: Int, highLimit: Int): Int {
        return (lowLimit .. highLimit).map{
            it.toString().sumBy { ch ->
                ch.toInt() - 48
            }
        }.groupingBy{ it }.eachCount().values.max()
    }
}